internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -263,3 +311,31 @@ func (r *Reconciler) reconcileExternal(ctx context.Context, clusterClass *cluste
 func uniqueObjectRefKey(ref *corev1.ObjectReference) string {
 	return fmt.Sprintf("Name:%s, Namespace:%s, Kind:%s, APIVersion:%s", ref.Name, ref.Namespace, ref.Kind, ref.APIVersion)
 }
+
+// extensionConfigToClusterClass maps an ExtensionConfigs with the corresponding ClusterClass to reconcile them on updates
+// of the ExtensionConfig.
+func (r *Reconciler) extensionConfigToClusterClass(o client.Object) []reconcile.Request {
+	result := []ctrl.Request{}
+
+	ext, ok := o.(*runtimev1.ExtensionConfig)
+	if !ok {
+		panic(fmt.Sprintf("Expected an ExtensionConfig but got a %T", o))
+	}
+
+	clusterClasses := clusterv1.ClusterClassList{}
+	selector, err := metav1.LabelSelectorAsSelector(ext.Spec.NamespaceSelector)
+	if err != nil {
+		return nil
+	}
+	if err := r.Client.List(
+		context.TODO(),
+		&clusterClasses,
+		client.MatchingLabelsSelector{Selector: selector},
+	); err != nil {
+		return nil
+	}
+	for _, ext := range clusterClasses.Items {
+		result = append(result, ctrl.Request{NamespacedName: client.ObjectKey{Name: ext.Name}})
```

Comment:
I think it would be also good if we additionally check if the ClusterClass is using this ExtensionConfig for variableDiscovery.


We can go all the way and build an index or add another check here. I think the index could work something like this:
```go
func ClusterClassByExtensionConfigName(o client.Object) []string {
	clusterClass, ok := o.(*clusterv1.ClusterClass)
	if !ok {
		panic(fmt.Sprintf("Expected Cluster but got a %T", o))
	}

	res := []string{}
	for _, patch := range clusterClass.Spec.Patches {
		if patch.External != nil && patch.External.DiscoverVariablesExtension != nil {
			// add namespaceselector handling
			
			split := strings.Split(*patch.External.DiscoverVariablesExtension, ".")
			res = append(res, split[0])
		}
	}
	return res
}
```

Reply:
(I can take it up as a follow-up if you want)

Reply:
I think the current version is correct - I'll add building an index as a follow up in the issue.